### DISCUSSION TITLE
All Our Project Are Belong to Octocat

### DISCUSSION THREAD

**Discussion by QuintillusCFC** [C7-Game/Prototype#449] [General]
Swinging back by C7 as I do every once in a blue moon, I saw that there is progress - amazing!  Things getting merged, code being written... almost makes me wish I didn't have all these Super Moderator projects I'm working on at this site called CivFanatics.  So I thought, "what's left for Carthage?", went to the Projects tab to check, and... nothing.

I'd seen that our projects (classic) would be migrated to projects (current) in August, but was definitely not aware that there was an exception that cards that hadn't been updated in a year would not be migrated.  Guess what?  The project had been idle for a bit over a year, so the result was _nothing was migrated_.  Thanks GitHub!

We weren't the only ones thrown for a loop; see this thread: https://github.com/orgs/community/discussions/137924

I poked around the Projects (Classic) API, which still exists until... November 19, 2024.  According to GitHub:

```
After the migration on August 23, you will still technically be able to use the REST or GraphQL API to update your classic Projects. We heavily discourage this as your updates will not be persisted to the migrated V2 Projects.

November 19, 2024: The [REST API for Projects (classic)](https://docs.github.com/en/rest/projects?apiVersion=2022-11-28) will be sunset.
```

So good news, we have until November 19 to save our projects.  The source data isn't all gone - they're in the Issues tab - but the sorting of them into projects is only accessible via the API.  As well as answering the question of "what's left for Carthage"? (although the only big features, I'm pretty sure, are save/load and Godot 4), there was some preliminary planning for Dutch and future milestones.

As you can see from that thread, I've done some poking around via the API to confirm that the data has not yet been deleted.

Not absolutely critical, but enough time went into creating it, and it added enough organization to the project, that I plan to at least take a stab as saving it.  I may create a Projects (New) out of it, but the goal is to at least not lose the data.

**Comment by pcen** [comment]
For now we should dump all of the project data as JSON and commit it to git if we don't have time to migrate all the issues to new projects format

**Comment by WildWeazel** [comment]
What if anything is left to be done to avoid losing data by 11/19? Is all of the raw data recovered?

**Reply by QuintillusCFC** [reply]
> What if anything is left to be done to avoid losing data by 11/19? Is all of the raw data recovered?

[C7 Project Status.txt](https://github.com/user-attachments/files/17402872/C7.Project.Status.txt)

Don't have time to commit it now but here is is... I think this is all of it (beyond column history)

**Reply by QuintillusCFC** [reply]
A more complete answer than I had time to write before dinner - I have not fully explored the API, so there may be a bit more than can be retrieved especially from that "status over time" standpoint, and who the card was assigned to, but that .txt file has all of the cards, their column and project.  The ones that are numerical entries correspond to issue IDs, and the ones that are text were just entered as free text on the project board, typically for Aztec or Babylon.

I may get back around to seeing if I can extract more info over the next few weeks.

**Comment by ajhalme** [comment]
I've had a go at importing the project data from the dump file `C7 Project Status.txt` into the org's project space:
https://github.com/orgs/C7-Game/projects?query=is%3Aopen

Onwards and upwards.
